src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -12761,7 +12891,21 @@ void Assembler::prefix(Register dst, Address adr, Prefix p) {
   }
 }
 
-void Assembler::prefix(Address adr) {
+void Assembler::prefix_rex2(Register dst, Address adr) {
+  if (adr.index_needs_rex2()) {
+    assert(false, "prefix(Register dst, Address adr) does not support handling of an X");
+  }
```

Comment:
this could be written as:
assert(!adr.index_needs_rex2(), "prefix(Register dst, Address adr) does not support handling of an X");

Reply:
Very nice.  Thank you, done.